src/subgraph/portfolio/index.ts

```diff
@@ -0,0 +1,168 @@
+      unrealizedPnl: totalUnrealizedPnl.toFixed(6),
+      realizedPnl: totalRealizedPnl.toFixed(6),
+    };
+  } catch (error) {
+    throw error;
+  }
+};
+
+const getDepositedCollateralBySnxAccount = (
+  collateralDeposits: collateralDepositsPortfolioData,
+  collateralPrice: { id: string; price: number }[],
+) => {
+  let totalCollateral = 0;
+  if (!collateralDeposits) return totalCollateral;
+
+  const depositedAmount = formatEther(collateralDeposits.depositedAmount ?? '0');
+
+  const collateralSymbol =
+    collateralMappingWithRegularSymbol.get(collateralDeposits.collateralSymbol?.toLowerCase() || '') ||
+    collateralDeposits.collateralSymbol?.toLowerCase();
+
+  const pythId = SYMBOL_TO_PYTH_FEED.get(collateralSymbol?.toUpperCase() || '');
+
+  // Check if pythId exists before trying to access collateralPrice
+  const price = pythId ? collateralPrice.find((p) => p.id === pythId.slice(2))?.price || 0 : 0;
+
+  //   console.log('price: ', collateralSymbol, pythId, price);
+
+  // Multiply by price and add to totalCollateral
+  totalCollateral = Number(depositedAmount) * price;
+
+  return totalCollateral;
+};
+
+const getUnRealizedPnlBySnxAccount = (
+  openPositionsArray: positionsPortolfio,
+  collateralPrice: { id: string; price: number }[],
+) => {
+  let totalUnrealizedPnl = 0;
+  if (!openPositionsArray) return totalUnrealizedPnl;
+  const size = formatEther(openPositionsArray.positionSize ?? '0');
+  const avgPrice = formatEther(openPositionsArray.avgPrice ?? '0');
+  const pythId = SYMBOL_TO_PYTH_FEED.get(openPositionsArray.market.marketSymbol?.toUpperCase() ?? '');
+  const price = pythId ? collateralPrice.find((p) => p.id === pythId.slice(2))?.price || 0 : 0;
+  //   console.log('currentMarketPrice', price);
+  const unrealizedPnL = (price - Number(avgPrice)) * Number(size);
+  totalUnrealizedPnl += unrealizedPnL;
+
+  return totalUnrealizedPnl;
+};
+
+const getRealizedPnlBySnxAccount = (closedPositionsArray: positionsPortolfio) => {
+  let totalRealizedPnl = 0;
+  if (!closedPositionsArray) return totalRealizedPnl;
+
+  const realizedPnl =
+    Number(closedPositionsArray.realizedPnl ?? '0') - Number(formatEther(closedPositionsArray.realizedFee ?? '0'));
+  totalRealizedPnl += realizedPnl;
+
+  return totalRealizedPnl;
+};
+
+const splitPositionsByStatus = (
+  positions: positionsPortolfio[],
+): { openPositions: positionsPortolfio[]; otherPositions: positionsPortolfio[] } => {
+  const openPositions: positionsPortolfio[] = [];
+  const otherPositions: positionsPortolfio[] = [];
+
+  positions.forEach((position) => {
+    if (position.status === 'OPEN') {
+      openPositions.push(position);
+    } else {
+      otherPositions.push(position);
+    }
+  });
+
+  return { openPositions, otherPositions };
+};
+
+export function transformPriceArray(priceArray: PriceObject[]): { id: string; price: number }[] {
+  return priceArray.map((obj) => ({
+    id: obj.id,
+    price: Number(obj.price.price) / 10 ** obj.price.expo,
```

Comment:
`price.expo` values are usually negative. Please use the absolute value of the field